dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -166,13 +169,28 @@ public ResponseEntity<String> listHandles(HttpServletRequest request, @PathVaria
 
     /**
      * Maps the handle to a correct response.
+     * If the metadata parameter is provided, return additional handle values.
      *
      * @param request        HttpServletRequest
      * @param handleResolver HdlResolverDTO - Handle resolver
-     * @return One element list using String if found, else null String.
+     * @return One element list using String if found, else map of metadata if param is entered,
+     * else "null" in case of an error
      */
     private String resolveToURL(HttpServletRequest request, HdlResolverDTO handleResolver) {
-        return mapAsJson(this.hdlResolverService.resolveToURL(ContextUtil.obtainContext(request), handleResolver));
+        String url = this.hdlResolverService.resolveToURL(ContextUtil.obtainContext(request), handleResolver);
+        String param = request.getParameter("metadata");
+        if (StringUtils.isBlank(param)) {
+            return mapAsJson(url);
+        }
+        String handle = handleResolver.getHandle();
+        try {
+            Map<String, String> metadata = HandlePlugin.getMapHandleValues(handle);
+            metadata.put("URL", url);
+            return mapAsJson(metadata);
+        } catch (HandleException e) {
+            log.error("Failed to resolve handle values for handle: " + handle, e);
+        }
+        return "null";
```

Comment:
Because in original implementation it was like "null" -> https://github.com/dataquest-dev/DSpace/pull/761/files#diff-48aa4c8ed31e44c4791f50fbd34fb735f4164bf59bb77427ec20c545d150296eR307